src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -1310,6 +1342,103 @@ internal string GetLeftPart(UriPartial part)
         }
 
         //
+        //
+        /// Transforms a character into its hexadecimal representation.
+        public static string HexEscape(char character)
+        {
+            if (character > '\xff')
+            {
+                throw new ArgumentOutOfRangeException(nameof(character));
+            }
+            char[] chars = new char[3];
+            int pos = 0;
+            UriHelper.EscapeAsciiChar(character, chars, ref pos);
+            return new string(chars);
```

Comment:
At some point (not in this PR), it'd be nice to look at refactoring this slightly to avoid the char[] allocation.